pangraph-0.2.0

thisiswhereitype opened the pull request:
This was to be patch 0.1.3 but I explored making the `Protograph` and changes to `Edge` which resulted in this needing a major bump. So workcraft will be 0.2.1.

pangraph-0.2.0 
* Addition of conversion and revert for FGL.
* Addition of Parser, Serializer and AST for GML.
* Change `Edge` types to now only construct with a `VertexID` type.
* Change `edgeEndpoints` to return VertexID.

Further internally there is now the following modules:
```haskell
-- A module to ease compatibility between Hexml and Pangraph.
Pangraph.Internal.HexmlExtra
-- A module containing boilerplate code for constructing of pangraphs.
Pangraph.Internal.ProtoGraph
```

thisiswhereitype committed:
Addition of Pangraph.Internal, containing Protograph.hs a module to aid in constructing Pangraphs from ASTs and HexmlExtra which contains functions which extend Hexml and/or interface Pangraph and Hexml namespaces. Further changes to Pangraph.FGL to work with new new API

thisiswhereitype committed:
FGL.revert implmentation and test

thisiswhereitype committed:
Merge remote-tracking branch 'upstream/master'
Returned to using Maybe Pangraph from Either for now.

thisiswhereitype committed:
Change let alignment in Pangraph.verifyGraph

thisiswhereitype committed:
Add changelog up to pangraph-0.1.2

thisiswhereitype committed:
Readme, changelog and cabal version for 0.2.0

thisiswhereitype committed:
Bump to lts-12.9 and move version dpendencies to stack.yaml from .cabal.

snowleopard commented on src/Pangraph/ProtoGraph.hs:
```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE OverloadedStrings #-}
```
It would be nice to add some top-level module documentation to describe the purpose of various modules.

snowleopard commented:
@thisiswhereitype What is the motivation behind `Protograph`? Isn't `Pangraph` sufficient? Having two seemingly similar datatypes in the library may be confusing. If `Protograph` is merely a convenient intermediate representation, it should perhaps live in an `Internal` module, with a warning that it is not a part of the API.

thisiswhereitype committed:
Moved ProtoGraph module internally, refactored it and its dependecies and improved documentation.

thisiswhereitype committed:
Improved documentaion

thisiswhereitype commented:
I have moved it as you suggest to an internal module and refactored. I have also improved the quality of the docs on all exposed modules. @snowleopard

snowleopard reviewed (approved):
Looks better now! Feel free to merge when ready.

thisiswhereitype commented:
Merged thanks for the review!
